utilities/transactions/pessimistic_transaction.cc

```diff
@@ -520,6 +520,8 @@ Status PessimisticTransaction::TryLock(ColumnFamilyHandle* column_family,
   } else {
     // If a snapshot is set, we need to make sure the key hasn't been modified
     // since the snapshot.  This must be done after we locked the key.
+    // TODO(myabandeh): If previously_locked then there is no need to validate
```

Comment:
@igorcanadi it seems that we can skip ValidateSnapshot and TrackKeys if it is already tracked. right?

Reply:
It seems that way to me, too.